pkg/ccl/streamingccl/streamproducer/stream_lifetime.go

```diff
@@ -226,30 +231,44 @@ func getReplicationStreamSpec(
 	if j.Status() != jobs.StatusRunning {
 		return nil, errors.Errorf("replication stream %d is not running", streamID)
 	}
+	details, ok := j.Details().(jobspb.StreamReplicationDetails)
+	if !ok {
+		return nil, errors.Errorf("job with id %d is not a replication stream job", streamID)
+	}
+	return buildReplicationStreamSpec(ctx, evalCtx, details.TenantID, false, details.Spans)
+
+}
+
+func buildReplicationStreamSpec(
+	ctx context.Context,
+	eval *eval.Context,
+	tenantID roachpb.TenantID,
+	forSpanConfigs bool,
+	targetSpans roachpb.Spans,
+) (*streampb.ReplicationStreamSpec, error) {
+	jobExecCtx := eval.JobExecContext.(sql.JobExecContext)
 
 	// Partition the spans with SQLPlanner
 	dsp := jobExecCtx.DistSQLPlanner()
 	planCtx := dsp.NewPlanningCtx(ctx, jobExecCtx.ExtendedEvalContext(),
 		nil /* planner */, nil /* txn */, sql.DistributionTypeSystemTenantOnly)
 
-	details, ok := j.Details().(jobspb.StreamReplicationDetails)
-	if !ok {
-		return nil, errors.Errorf("job with id %d is not a replication stream job", streamID)
-	}
-	replicatedSpans := details.Spans
-	spans := make([]roachpb.Span, 0, len(replicatedSpans))
-	for _, span := range replicatedSpans {
-		spans = append(spans, *span)
-	}
-	spanPartitions, err := dsp.PartitionSpans(ctx, planCtx, spans)
+	spanPartitions, err := dsp.PartitionSpans(ctx, planCtx, targetSpans)
 	if err != nil {
 		return nil, err
 	}
 
+	var spanConfigsStreamID streampb.StreamID
+	if forSpanConfigs {
+		spanConfigsStreamID = streampb.StreamID(builtins.GenerateUniqueInt(builtins.ProcessUniqueID(eval.NodeID.SQLInstanceID())))
+	}
+
 	res := &streampb.ReplicationStreamSpec{
-		Partitions:     make([]streampb.ReplicationStreamSpec_Partition, 0, len(spanPartitions)),
-		SourceTenantID: details.TenantID,
+		Partitions:         make([]streampb.ReplicationStreamSpec_Partition, 0, len(spanPartitions)),
+		SourceTenantID:     tenantID,
+		SpanConfigStreamID: spanConfigsStreamID,
```

Comment:
What is this streamID going to be used for now that we've decided we don't need a producer-side job?